### PR TITLE
[tests-only][full-ci]added then steps for apiProvisioningGroups-v1 suite

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This PR adds ``` Then ``` steps to existing API test scenarios to better check the results of ``` When ``` steps.

### Suite Covered
- ```apiProvisioningGroups-v1```

## Related Issues
- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?
- Locally
- CI

## Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by kiranparajuli589 — COMMENTED** [review]
looks good

File: tests/acceptance/features/apiProvisioningGroups-v1/addGroup.feature [hunk 1]
```diff
@@ -13,14 +13,14 @@ Feature: add groups
       | groupname   | comment                               |
       | simplegroup | nothing special here                  |
       | España§àôœ€ | special European and other characters |
-      | नेपाली        | Unicode group name                    |
+      | नेपाली      | Unicode group name                    |
```
Comment (grgprarup) [inline tests/acceptance/features/apiProvisioningGroups-v1/addGroup.feature:16]: Please fix the alignment

File: tests/acceptance/features/apiProvisioningGroups-v1/addGroup.feature [hunk 2]
```diff
@@ -13,14 +13,14 @@ Feature: add groups
...
     Then the OCS status code of responses on all endpoints should be "100"
     And the HTTP status code of responses on all endpoints should be "200"
     And these groups should exist:
       | groupname   |
       | simplegroup |
       | España§àôœ€ |
-      | नेपाली        |
+      | नेपाली      |
```
Comment (grgprarup) [inline tests/acceptance/features/apiProvisioningGroups-v1/addGroup.feature:23]: Please fix the alignment

File: tests/acceptance/features/apiProvisioningGroups-v1/getUserGroups.feature [hunk 3]
```diff
@@ -107,16 +107,16 @@ Feature: get user groups
...
     When the administrator gets all the groups of user "brand-new-user" using the provisioning API
-    Then the groups returned by the API should be
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And the groups returned by the API should be
       | brand-new-group      |
       | 0                    |
       | Admin & Finance (NP) |
       | admin:Pokhara@Nepal  |
       | नेपाली               |
```
Comment (grgprarup) [inline tests/acceptance/features/apiProvisioningGroups-v1/getUserGroups.feature:117]: Please fix the alignment

**Comment by kiranparajuli589** [comment]
feature indentations can be fixed separately. here is a ticked for reminder https://github.com/owncloud/core/issues/39852

**Review by phil-davis — APPROVED** [review]

**Pull request merged**